app/src/main/java/fr/free/nrw/commons/nearby/NearbyActivity.java

```diff
@@ -74,19 +94,23 @@ public boolean onOptionsItemSelected(MenuItem item) {
             case R.id.action_refresh:
                 refreshView();
                 return true;
-            case R.id.action_map:
-                showMapView();
-                if (isMapViewActive) {
-                    item.setIcon(R.drawable.ic_list_white_24dp);
-                } else {
-                    item.setIcon(R.drawable.ic_map_white_24dp);
-                }
+            case R.id.action_toggle_view:
+                toggleView();
+                toggleIcon(item);
                 return true;
             default:
                 return super.onOptionsItemSelected(item);
         }
     }
 
+    private void toggleIcon(MenuItem item) {
```

Comment:
This can be one line of code - if you use a ternary expression, or if you added a 'getIcon()' to the enum and initialized the two values with resource ID integers.